File: sdk/src/wallet/bindings/nodejs/examples/getting-started.js
const { AccountManager, CoinType } = require('@iota/wallet');

const NODE_URL = 'https://api.testnet.shimmer.network'
const STORAGE_PATH = 'walletdb'
const STRONGHOLD_SNAPSHOT_PATH = 'vault.stronghold'

async function main() {
    // Change to a secure password.
    let password = 'some-secure-password'

    // Set up and store the wallet.
    const accountManagerOptions = {
        storagePath: STORAGE_PATH,
        clientOptions: {
            nodes: [NODE_URL],
            localPow: true,
        },
        coinType: CoinType.Shimmer,
        secretManager: {
            stronghold: {
                snapshotPath: STRONGHOLD_SNAPSHOT_PATH,
                password: password,
            },
        },
    };

    const manager = new AccountManager(accountManagerOptions);

    // Generate a mnemonic and store it in the Stronghold vault.
    const mnemonic = await manager.generateMnemonic();
    await manager.storeMnemonic(mnemonic);

    // Create an account and get the first address.
    const account = await manager.createAccount({
        alias: 'Alice',
    });
    const address = await account.addresses().then(addresses => addresses[0]);

    // Print the account data.
    console.log(`Mnemonic:\n${mnemonic}\n`);
    console.log(`Address:\n${address.address}\n`);

    process.exit(0);
}

main();

File: sdk/src/wallet/bindings/python/examples/getting-started.py
# Copyright 2023 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from iota_wallet import IotaWallet, StrongholdSecretManager

# A name to associate with the created account.
ACCOUNT_ALIAS = 'Alice'

# The node to connect to.
NODE_URL = 'https://api.testnet.shimmer.network'

# The cointype to use.
SHIMMER_COIN_TYPE = 4219

# A password to encrypt the stored data.
# WARNING: Never hardcode passwords in production code.
STRONGHOLD_PASSWORD = 'a-secure-password'

# The path to store the account snapshot.
STRONGHOLD_SNAPSHOT_PATH = 'vault.stronghold'

# Setup Stronghold secret manager
secret_manager = StrongholdSecretManager(STRONGHOLD_SNAPSHOT_PATH, STRONGHOLD_PASSWORD)

# Set up and store the wallet.
client_options = {
    'nodes': [NODE_URL],
}

wallet = IotaWallet(
    client_options=client_options,
    coin_type=SHIMMER_COIN_TYPE,
    secret_manager=secret_manager
)

# Generate a mnemonic and store it in the Stronghold vault.
# INFO: It is best practice to back up the mnemonic somewhere secure.
mnemonic = wallet.generate_mnemonic()
wallet.store_mnemonic(mnemonic)

# Create an account.
wallet.create_account('Alice')

# Get the first address and print it.
address = wallet.get_account('Alice').addresses()[0]
print(f'Address:\n{address["address"]}\n')
